libraries/botbuilder-lg/README.MD

```diff
@@ -0,0 +1,188 @@
+
+Then make sure you include the platform specific language generation library.
+
+For C#, add Microsoft.Bot.Builder.LanguageGeneration.
+For NodeJS, add botbuilder-lg
+
+Load the template manager with your .lg file(s)
+
+For C#
+
+```c#
+    // multi lg files
+    TemplateEngine lgEngine = new TemplateEngine().AddFiles(filePaths, importResolver?);
+
+    // single lg file
+    TemplateEngine lgEngine = new TemplateEngine().AddFile(filePath, importResolver?);
+```
+
+For NodeJS
+
+```typescript
+    // multi lg files
+    let lgEngine = new TemplateEngine.addFiles(filePaths, importResolver?);
+
+    // single lg file
+    let lgEngine = new TemplateEngine.addFile(filePath, importResolver?);
+```
+
+When you need template expansion, call the templateEngine and pass in the relevant template name
+
+For C#
+
+```c#
+    await turnContext.SendActivityAsync(lgEngine.EvaluateTemplate("<TemplateName>", entitiesCollection));
+```
+
+For NodeJS
+
+```typescript
+    await turnContext.sendActivity(lgEngine.evaluateTemplate("<TemplateName>", entitiesCollection));
+```
+
+If your template needs specific entity values to be passed for resolution/ expansion, you can pass them in on the call to `evaluateTemplate`
+
+For C#
+
+```c#
+    await turnContext.SendActivityAsync(lgEngine.EvaluateTemplate("WordGameReply", new { GameName = "MarcoPolo" } ));
+
+```
+
+For NodeJS
+
+```node
+    await turnContext.sendActivity(lgEngine.evaluateTemplate("WordGameReply", { GameName = "MarcoPolo" } ));
+```
+
+## Multi-lingual generation and language fallback policy
+Quite often your bot might target more than one spoken/ display language. To help with resource management as well as implement a default language fall back policy, you can either use `MultiLanguageGenerator` or `ResourceMultiLanguageGenerator`. See [here][25] for an example.
+
+## Grammar check and correction
+
+The current library does not include any capabilities for grammar check or correction.
+
+## Packages
+
+Packages for C# are available under the [BotBuilder MyGet feed][12]
+
+## Change Log
+### 4.6 PREVIEW
+- \[**NEW**\] [VS code extension][22] for LG (syntax highlighting, auto-suggest (including expressions, pre-built functions, template names etc), validation)
+- LG file format:
+    - Support for [Switch..Case..Default][20]
+    - Support for [import reference][21] to another .lg file.
+- [API changes][2]: 
+    - Dropped FromFile and FromText methods in favor of AddFile and AddFiles. 
+    - Added ability to provide a delegate to externally resolve import references found in content. 
+- \[**NEW**\] Translate functionality in [MSLG CLI][23]
+
+### 4.5 PREVIEW
+- Initial preview release
+
+[1]:https://github.com/Microsoft/BotBuilder/blob/master/specs/botframework-activity/botframework-activity.md
+[2]:./docs/api-reference.md
```

Comment:
These links are not going to work unless we are moving all the docs over which we should not do.